### PR TITLE
mon/OSDMonitor: make snaps on tier pool should not be allowed

### PR DESCRIPTION
**Pull request by LiumxNL**
Signed-off-by: Mingxin Liu mingxin@xsky.com

### CODE REVIEW

**Comment by dachary** [comment]
jenkins test this please (jenkins hung)

**Comment by LiumxNL** [comment]
@tchaikov please review, thanks!

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -7891,7 +7899,7 @@ bool OSDMonitor::preprocess_pool_op(MonOpRequestRef op)
     }
     return false;
   case POOL_OP_DELETE_SNAP:
-    if (p->is_unmanaged_snaps_mode()) {
+    if (p->is_unmanaged_snaps_mode() || p->is_tier()) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: it's not necessary to check if `p->is_tier()` again? if it is not allowed to create a snap for a tier pool?

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -7008,6 +7012,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       ss << "pool " << poolstr << " snap " << snapname << " does not exist";
       err = 0;
       goto reply;
+    } else if (p->is_tier()) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: ditto.

**Comment by LiumxNL** [comment]
@tchaikov updated.

**Pull request merged**